pdb-alignment-pipeline/src/main/java/org/cbioportal/pdb_annotation/scripts/PdbScriptsPipelineRunCommand.java

```diff
@@ -0,0 +1,491 @@
+	}
+	
+	/**
+	 * generate wget command
+	 * 
+	 * @param urlFilename
+	 * @param localFilename
+	 * @return
+	 */
+	private List<String> makeDownloadCommand(String urlFilename, String localFilename) {
+		List<String> list = new ArrayList<String>();
+		// Building the following process command
+		list.add("wget");
+		list.add("-O");
+		list.add(localFilename);
+		list.add(urlFilename);
+		return list;
+	}
+	
+	/**
+	 * download files from urlFilename from Internet to localFilename of the system
+	 * 
+	 * @param urlFilename
+	 * @param localFilename
+	 * @return
+	 */
+	public boolean downloadfile(String urlFilename, String localFilename) {
+			try {
+				System.out.println("[SHELL] Download file "+urlFilename+" ...");
+				ProcessBuilder dbBuilder = new ProcessBuilder(makeDownloadCommand(urlFilename,localFilename ));
+				Process makeDB = dbBuilder.start();
+				makeDB.waitFor();
+				System.out.println("[SHELL] "+urlFilename+" completed");
+			} catch (Exception ee) {
+				System.err.println("[SHELL] Fatal Error: Could not Successfully download files");
+				ee.printStackTrace();
+			}	
+		return true;
+	}
+	
+	/**
+	 * read FTP file to list
+	 * 
+	 * @param urlStr
+	 * @return
+	 */
+	public List<String> readFTPfile2List(String urlStr){
+		List<String> list = new ArrayList();
+		try{
+			URL url = new URL(urlStr);
+	        URLConnection con = url.openConnection();
+	        BufferedReader in = new BufferedReader(new InputStreamReader(
+	                                    con.getInputStream()));
+	        String inputLine;
+	        while ((inputLine = in.readLine()) != null) 
+	        	list.add(inputLine);
+	        in.close();
+			
+		}catch(Exception ex){
+			ex.printStackTrace();
+		}
+		return list;
+	}
+	
+	/**
+	 * read FTP file to String
+	 * @param urlStr
+	 * @return
+	 */
+	public String readFTPfile2Str(String urlStr){
+		String str ="";
+		try{
+			URL url = new URL(urlStr);
+	        URLConnection con = url.openConnection();
+	        BufferedReader in = new BufferedReader(new InputStreamReader(
+	                                    con.getInputStream()));
+	        String inputLine;
+	        while ((inputLine = in.readLine()) != null) 
+	        	str=str+inputLine+"\n";
+	        in.close();
+			
+		}catch(Exception ex){
+			ex.printStackTrace();
+		}
+		return str;
+	}
+	
+	/**
+	 * prepare weekly updated PDB files
+	 * 
+	 * @param currentDir
+	 * @param updateTxt
+	 * @param delPDB
+	 * @return
+	 */
+	public List<String> prepareUpdatePDBFile(String currentDir, String updateTxt, String delPDB){
+		List<String> listOld=new ArrayList<String>();
+		try{
+			System.out.println("[SHELL] Weekly Update: Create deleted list");
+			
+			
+			FileUtils.forceMkdir(new File(currentDir));
+			
+			String addFileName = currentDir+updateTxt;
+			File addFastaFile = new File(addFileName);
+			
+			String delFileName = currentDir+delPDB;
+			
+			List listAdd = readFTPfile2List("ftp://ftp.wwpdb.org/pub/pdb/data/status/latest/added.pdb"); 
```

Comment:
@juexinwang instead of hard-coding the data/file sources, can we make all the URLs configurable?